src/beliefpropagation/beliefpropagation.jl

```diff
@@ -135,9 +174,10 @@ function belief_propagation(
   subgraph_vertices=nothing,
   niters=20,
   target_precision::Union{Float64,Nothing}=nothing,
```

Comment:
```suggestion
  target_precision=nothing,
```
I generally don't put type assertions on keyword arguments.

Reply:
Again, to make things more generic, for example maybe you want to use single precision in which case it makes sense to pass something that is `Float32`.